### PR TITLE
Complete Project Setup - Budget App

### PR DESCRIPTION
**Pull request by ouasamine**
## Overview
This is a complete project creation and setup operation, I used Rails to create an app that will display transactions made by a registered user grouped in categories.

## Implemented Features
- User registration and authentications system using Devise gem
- A home page that shows all the categories made by the signed-in user
- Clicking on a category lead to a page displaying all the transactions within that category
- The home page (categories) has a button that leads to a form for adding new categories
- The transactions page has a button that leads to a form for adding new transactions

**Here is a [video](https://www.loom.com/share/88f3a758b7864c81a8882eaa21ff25c2) presentation of the final version of the app**

### CODE REVIEW

File: app/views/devise/shared/_links.html.erb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<%- if controller_name != 'sessions' %>
+  <%= link_to "Log in", new_session_path(resource_name) %><br />
+<% end %>
+
+<%- if devise_mapping.registerable? && controller_name != 'registrations' %>
+  <%= link_to "Sign up", new_registration_path(resource_name) %><br />
```
Comment (t-yanick) [inline app/views/devise/shared/_links.html.erb:6]: - [ ] When I want to register a new user, I get an error message, but I am able to use same details to log in . See screenhot below. With a video presentation I would have ascertain whether it is working at your end or not.
Screen Shot 2023-02-17 at 5 07 46 AM

You have a user model and no user controller, but you rather have home controller and no home model. Just check to ensure that this cannot be the origin of this issue.

File: app/views/groups/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  <h1>Categories</h1>
+  <%= button_to image_tag("exit.png"), destroy_user_session_path, method: :delete, class: "butt-icon" %>
+</header>
+
+<section>
+  <div class="cards-container">
+    <% @groups.each do |group| %>
+    <a class="card-link-wrapper" href="<%= group_transacs_path(group) %>">
+      <div class="card">
+        <div class="img-plchldr column"><img src="<%= group.icon %>" alt="Transaction icon"></div>
```
Comment (t-yanick) [inline app/views/groups/index.html.erb:11]: - [ ] The text Transaction icon is overriding the name of the transaction and I don't think it is very necessary. Just the Icon is necessary. 
- [ ] Also your typogragraphy is not exact like the original design. Kindly review that.
Screen Shot 2023-02-17 at 5 22 06 AM

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️

Hi @ouasamine , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Spec tests added
:heavy_check_mark: Migrations with correct data types
:heavy_check_mark: Models have correct associations as per E-R diagram
:heavy_check_mark: Main functionalities working as expected
:heavy_check_mark: All pages as per project requirement present
:heavy_check_mark: Neccesary routes added with root route specified
:heavy_check_mark: PR description and ReadMe are professional

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ] Great job working on this capstone. You missed a major requirement with regards to documentation. You were supposed to make a video presentation of maximum 5 minutes in which you present the project features and a piece of code which you are proud of or find interesting. Kindly make the video and put the link in your ReadMe.
Screen Shot 2023-02-17 at 4 56 21 AM


_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by t-yanick** [comment]
- [ ] I am just seeing the link to your video presentation in the PR description. Kindly put the link also in your ReadMe

**Comment by ouasamine** [comment]
> * [ ]  I am just seeing the link to your video presentation in the PR description. Kindly put the link also in your ReadMe

@t-yanick  is it part of the requirements to put it in the README?

**Comment by t-yanick** [comment]
@ouasamine I think a User of your App will easily access your ReadMe rather than the PR description. I think having it in the ReadMe is better. The Video is part of documentation and I think documentation is best fit to be in the ReadMe

**Comment by ouasamine** [comment]
> @ouasamine I think a User of your App will easily access your ReadMe rather than the PR description. I think having it in the ReadMe is better. The Video is part of documentation and I think documentation is best fit to be in the ReadMe

@t-yanick I belive this should have had an [OPTIONAL] tag since it's not part of the core requirement, I have added it to the README anyway, good luck.

**Comment by t-yanick** [comment]
@ouasamine as I explained earlier, I missed the link in the PR description and only saw it after submitting the review feedback.  My error and I am sorry about that.

The feedback with regards to the link was not where it was found, but it being absent since I missed it.  Sorry once and Happy coding as you move ahead.

**Review by Davidosky007 — APPROVED** [review]
Hi @ouasamine 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* Professional video presentation 👍 
* No linter errors 👍 
* All tests passed ✔️ 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**